core/src/test/java/io/neow3j/protocol/core/ResponseTest.java

```diff
@@ -1637,7 +1638,32 @@ public void testInvokeFunction_withoutOrEmpty_Params() {
         assertThat(invokeFunction.getInvocationResult(), is(notNullValue()));
         assertThat(invokeFunction.getInvocationResult().getScript(),
                 is("10c00c0962616c616e63654f660c14897720d8cd76f4f00abfa37c0edd889c208fde9b41627d5b52"));
-        assertThat(invokeFunction.getInvocationResult().getState(), is("FAULT"));
+        assertThat(invokeFunction.getInvocationResult().getState(), is(NeoVMStateType.FAULT));
+        assertThat(invokeFunction.getInvocationResult().getGasConsumed(), is("2007390"));
+        assertThat(invokeFunction.getInvocationResult().getStack(), is(notNullValue()));
+        assertThat(invokeFunction.getInvocationResult().getStack(), hasSize(0));
+    }
+
+    @Test
+    public void testInvokeFunction_emptyState() {
+        buildResponse(
+                "{\n" +
+                        "    \"jsonrpc\": \"2.0\",\n" +
+                        "    \"id\": 1,\n" +
+                        "    \"result\": {\n" +
+                        "        \"script\": \"10c00c0962616c616e63654f660c14897720d8cd76f4f00abfa37c0edd889c208fde9b41627d5b52\",\n" +
+                        "        \"state\": \"\",\n" +
```

Comment:
Can this case really occur?

Reply:
Not sure, but I think that's the reason why they have `NONE`. 😄 

I would prefer to leave this test like this, even if it's not realistic. It tests the serialization/deserialization of the JSON to `NONE` state. 👍